core/src/main/scala/com/banno/kafka/producer/ProducerImpl.scala

```diff
@@ -106,15 +107,16 @@ case class ProducerImpl[F[_], K, V](p: Producer[K, V])(implicit
     * proceeding, you should call sendSync or sendAsync.
     */
   def sendAndForget(record: ProducerRecord[K, V]): F[Unit] =
-    F.delay(sendRaw(record))
+    F.interruptible(sendRaw(record))
       .void // discard the returned JFuture[RecordMetadata]
 
   /** Like `sendSync`, but blocks a compute thread awaiting the acknowledgement.
     * Do not use.
     */
   @deprecated("Use sendAsync, or send in kafka4s-6.x", "5.0.4")
   def sendSync(record: ProducerRecord[K, V]): F[RecordMetadata] =
-    F.delay(sendRaw(record)).flatMap(jFut => F.interruptible(jFut.get()))
+    F.delay(sendRaw(record))
```

Comment:
Isn't this one interruptible?  Doesn't `sendRaw(record)` block if the internal buffer is full?

Reply:
Ah yeah, Ross is right. We'll be getting rid of `sendSync` at some point, but should make it work better until then.

Reply:
copy that.  on it! ty.